### ISSUE TITLE
Find-My-Friends (fmf) only returning 1 device to track

### ISSUE THREAD

**Issue by SketchCND** [gcobb321/icloud3#176]
Following iOS 15 update, iCloud3 Find-My-Friends (fmf) is only detection 1 of 2 iPhones that were previously working. I have updated to the 2.4.4 release but the problem persists. I've checked all the device names etc and they are unchanged from before the upgrade and location sharing for the iOS FindMy app is tracking both devices correctly.

Here is excerpt from iCloud even log:

<><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
	Initializing iCloud3 v2.4.4 > Complete, Took 16.42 sec

...

11:47:21					
	Tracking Devices > CRLF• sketch_iphone > Craig, iOSAppSuffix-, TrkMethod-FmF
11:47:20					
	Not Tracking Device > Rachel (rachel_iphone)

1:47:20					
	iCloud3 Error for Rachel (rachel_iphone)/rachel_iphone > The iCloud Account for s******t@****.com did not return any device information for this device.CRLF-------------------------CRLF 1. Restart iCloud3 on the Event_log screen or restart HA.CRLF 2. Verify the devicename on the `devices/device_name: rachel_iphone` parameter if the error persists.CRLF 3. Refresh the Event Log in your browser to refresh the list of devices.

...
	Initializing iCloud3 v2.4.4 > Wednesday, Oct 06

<><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>

Any ideas where I could continue troubleshooting?

Thanks

**Comment by gcobb321** [comment]
When iCloud3 starts, the Event Log (Stage 2) lists the phones in the FamShr list and in the Find-My-Friends app. For FmF, it shows the email address for each phone found and the phone's name. The FamShr list shows the phone's name. Check the Event Log and see the phones that were returned from your iCloud account to see if anything is unusual. If the phone is not listed, you may need to add it back to the FindMy app. Also see if both phones can see each other in the FindMy app. If it can not see each other, iCloud3 won't either.

**Comment by SketchCND** [comment]
Gary,

Thanks for the pointers. I checked the Stage 2 log portion and indeed the 2nd device was not listed there. My configuration was correct with the approriate device names and email address. Using FindMy app the both devices could locate/track one another without issue. I stoppped sharing from both devices and readded them in the FindMy app and again they could locate/track each other. I restarted iCloud3 but the problem remained. 
I then  renamed the iPhone (in Settings-->About) and updated my configs accordingly. After iCloud3 restart, I now have both devices being tracked. Ultimately not sure why this would have fixed it but nonetheless it did. 

Thanks again for taking the time to respond to my initial comments!
